_posts/2021-05-22-report-arms-spring2021.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: post
+title:  "Report: ARMS - reducing robot simulation code by 92%"
+author: "Sean O'Connor"
+categories: [ "robotics", "simulation", "gazebo", "sdf", "javascript"  ]
+image: assets/images/arms1/robot_image.png
+---
+*A concise and intuitive alternative for writing verbose robotics simulation languages, built with browser JavaScript.  For more, check out my [blog](https://oapostrophe.github.io/) and [GitHub](https:/github.com/oapostrophe).*
+
+## Project Background
+
+Let's say you're designing a robot.  This robot will be used to go over rough terrain as part of post-disaster search-and-rescue missions, so it's important that it be both tall enough to clear uneven ground and stable enough not to topple over on bumps.  You want to know: what height and width will offer the optimal combination of ground clearance and stability?
+
+One possible way to find out might be to build several different robots of varying dimensions and test how they perform.  But building each physical robot is expensive and time-consuming, not to mention that there might be a lot of feasible height-weight combinations.  So you also want to know: is there a way we can test different builds, without the cost of physically building every configuration we want to test?
+
+![Selife taken by a robot from the ARCs lab during its construction](../assets/2021-05-22-report-arms-spring2021/robot_selfie.png)  
+
+*One of our lab robots taking a selfie during its construction.*
+
+The answer is simulation.  By simulating each potential robot design, we can test how it performs without wasting time or materials to physically build it.  Simulation has an important place in the robotics field, and a number of tools exist for the simulations of robots: [DART](https://dartsim.github.io/), [Gazebo](http://gazebosim.org/), and [URDF](http://wiki.ros.org/urdf) to name a few.  But one common issue with such simulations is that many of these tools require verbose and difficult-to-read input.  For instance, let's look at the XML-based [SDFormat](http://sdformat.org/) used by Gazebo.  Consider one of the simplest possible objects - a 1x1x1 cube:
+
+![plain cube](../assets/2021-05-22-report-arms-spring2021/gazebo_cube.png)
+
+*A gray cube on a gray background in Gazebo - so basic, you almost can't see it.*
+
+Below is the SDF code required to create this basic object:
+
+![1x1x1 cube in SDF](../assets/2021-05-22-report-arms-spring2021/sdf_cube_snippet.png)
+
+Look at all those tags!  I'm getting a headache already.  Wouldn't it be nice if instead we could write something like this:
+```
```

Comment:
Add a space before the back-ticks and the paragraph above. Also, give the style as text.

Reply:
What do you mean by giving the style as text, like this?

```text
(code here)
```